src/coord/src/catalog.rs

```diff
@@ -193,6 +186,17 @@ impl CatalogState {
         &self.by_id[id]
     }
 
+    /// Returns all indexes on this object known in the catalog.
+    pub fn get_indexes_on(&self, id: GlobalId) -> impl Iterator<Item = (GlobalId, &Index)> {
+        self.get_by_id(&id)
+            .used_by()
+            .iter()
+            .filter_map(move |uses_id| match self.get_by_id(uses_id).item() {
+                CatalogItem::Index(index) if index.on == id => Some((*uses_id, index)),
+                _ => None,
+            })
+    }
```

Comment:
This is mostly just moved from below, btw. Moved out of `Catalog` and into `CatalogState`. The return value has been tweaked slightly too, to return the index metadata alongside the ID.